File: src/main/java/br/com/jiratorio/domain/vo/ChartAggregator.java
package br.com.jiratorio.domain.vo;

import br.com.jiratorio.domain.embedded.Chart;
import br.com.jiratorio.domain.embedded.ColumnTimeAvg;
import br.com.jiratorio.domain.embedded.Histogram;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChartAggregator {

    private Histogram histogram;

    private Chart<String, Long> estimated;

    private Chart<String, Double> leadTimeBySystem;

    private Chart<String, Long> tasksBySystem;

    private Chart<String, Double> leadTimeBySize;

    private List<ColumnTimeAvg> columnTimeAvg;

    private Chart<String, Double> leadTimeByType;

    private Chart<String, Long> tasksByType;

    private Chart<String, Double> leadTimeByProject;

    private Chart<String, Long> tasksByProject;

    private Chart<String, Double> leadTimeCompareChart;

    private Chart<String, Double> leadTimeByPriority;

    private Chart<String, Long> throughputByPriority;

    private List<DynamicChart> dynamicCharts;

}

File: src/main/kotlin/br/com/jiratorio/domain/chart/ChartAggregator.kt
package br.com.jiratorio.domain.chart

import br.com.jiratorio.domain.dynamicfield.DynamicChart
import br.com.jiratorio.domain.entity.embedded.Chart
import br.com.jiratorio.domain.entity.embedded.Histogram
import br.com.jiratorio.domain.entity.embedded.IssueProgression

data class ChartAggregator(

    val histogram: Histogram,

    val leadTimeByEstimate: Chart<String, Double>,

    val throughputByEstimate: Chart<String, Int>,

    val leadTimeBySystem: Chart<String, Double>,

    val throughputBySystem: Chart<String, Int>,

    val leadTimeByType: Chart<String, Double>,

    val throughputByType: Chart<String, Int>,

    val leadTimeByProject: Chart<String, Double>,

    val throughputByProject: Chart<String, Int>,

    val leadTimeByPriority: Chart<String, Double>,

    val throughputByPriority: Chart<String, Int>,

    val leadTimeCompareChart: Chart<String, Double>,

    val dynamicCharts: List<DynamicChart>,

    val issueProgression: IssueProgression

)
